source/agora/utils/Test.d

```diff
@@ -624,62 +627,274 @@ unittest
-    result.outputs[0].value.mustAdd(remainder);
-    const inputHash = input.hashFull();
+        Params:
+            address = finding `PublicKey`
+            unknowns =  unknown `KeyPair`s
 
-    // Add support for Transactions with multiple recipients
-    foreach (idx, const ref _; input.outputs)
-        result.inputs ~= Input(inputHash, cast(uint) idx);
+        Returns:
+            returns KeyPair.init if `address` is not found.
 
-    scope sign = (KeyPair kp, Hash h) @trusted { return kp.secret.sign(h[]); };
+    ***************************************************************************/
 
-    const resultHash = result.hashFull();
-    foreach (idx, ref in_; result.inputs)
+    private KeyPair getOwnerKeyPair (scope const PublicKey address,
+        scope const(KeyPair)[] unknowns = null) @safe
     {
-        auto rng = from.find!(a => a.address == input.outputs[idx].address);
-        assert(rng.length);
-        const owner = rng[0];
-        in_.signature = sign(owner, resultHash);
+        version(unittest)
+        {
+            auto gen_keypair = () @trusted {
+                return getGenesisKeyPair();
+            }();
+
+            if (address == gen_keypair.address)
+                return gen_keypair;
+        }
+        if (unknowns.length > 0)
+        {
+            auto index = unknowns.find!(res => res.address == address);
+            if (!index.empty)
+                return index.front;
+        }
+        return WK.Keys[address];
+    }
+
+    /***************************************************************************
+
+        Splits the attached input into multiple outputs.
+
+        Params:
+            toward = address of destination to be transferred.
+
+    ***************************************************************************/
+
+    public void split (scope const PublicKey[] toward...) @safe
+    {
+        if ((this.inputs.length == 0) || (toward.length == 0))
+            return;
+
+        Amount amount;
+        if (!this.inputs.all!(val => amount.add(val.output.value)))
+            assert(0, "Amount overflow");
+        auto remainder = amount.div(toward.length);
+
+        this.data.outputs = null;
+        foreach (key; toward)
+            this.data.outputs ~= Output(amount, key);
+
+        this.data.outputs[0].value.mustAdd(remainder);
+    }
+
+
+    /***************************************************************************
+
+        Distributes the attached input into multiple outputs with
+        the same amount.
+
+        Params:
+            share = amount to be distributed to one address
+            use_remainder = whether to use the remaining amount
+                after distribution.
+            toward = address of destination to be transferred.
+
+    ***************************************************************************/
+
+    public void distribute (scope const Amount share, bool use_remainder,
+        scope const PublicKey[] toward...) @safe
```

Comment:
This offers a rather confusing API:
- In general, we want to offer a simple API that covers most needs. Users can always manually modify the tx being built in order to satisfy their specific needs;
- As a result, `use_remainder` is the first thing that stands out. Remove it;
- Likewise, why is `share` needed ? The idea is that if you want to build complex scenarii, you will chain calls to `merge` and `split`.

Reply:
I will remove